### PR TITLE
Replace uglify with terser plugin

### PR DESCRIPTION
**Pull request by jaredpalmer**
- Replace `rollup-plugin-uglify` with `rollup-plugin-terser`

cc @TrySound

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -60,14 +60,17 @@ const buildUmd = ({ env }) => ({
...
+        sourcemap: true,
         output: { comments: false },
         compress: {
           keep_infinity: true,
           pure_getters: true,
         },
         warnings: true,
+        ecma: 5,
         toplevel: false,
+        mangle: false,
```
Comment (TrySound) [inline rollup.config.js]: Why not mangle? This is expected from production umd I think.

Comment (jaredpalmer) [reply rollup.config.js]: What options should we use for mangle?

Comment (TrySound) [reply rollup.config.js]: No options. Just remove `mangle: false`. Or is there a reason for it?

Comment (jaredpalmer) [reply rollup.config.js]: No idea. Just wanted to check.

File: rollup.config.js [hunk 2]
```diff
@@ -60,13 +60,15 @@ const buildUmd = ({ env }) => ({
...
+      terser({
+        sourcemap: true,
         output: { comments: false },
         compress: {
           keep_infinity: true,
           pure_getters: true,
         },
         warnings: true,
+        ecma: 5,
         toplevel: false,
```
Comment (jaredpalmer) [inline rollup.config.js]: @TrySound should we run terser on CJS production build as well? I see that it can accept a `format` option.

Comment (TrySound) [reply rollup.config.js]: Yeah, probably

**Comment by jaredpalmer** [comment]
Moved to #1261

**Pull request closed without merging**